### PR TITLE
azure- resource - Added support for hdinsight resource

### LINKED ISSUE #4530: azure - add HDInsight resource

### PR DESCRIPTION
**Pull request by wsf11**
Addresses part 1 of issue #3311 and closes #4530 . The next step is to add a resize action for HDInsight. 

Policy to run

         policies:
          - name: hdinsight-policy
            resource: azure.hdinsight
            filters:
              - type: value
                key: properties.clusterDefinition.kind
                value_type: normalize
                value: hadoop

Hadoop can be changed with other values that Azure HDInsight supports such as Spark, Storm, or Kafka.

### CODE REVIEW

File: tools/c7n_azure/tests/templates/hdinsight.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                                    "username": "[parameters('sshUserName')]",
+                                    "password": "[parameters('sshPassword')]"
+                                }
+                            }
+                        },
+                        {
+                            "name": "workernode",
+                            "targetInstanceCount": 4,
+                            "hardwareProfile": {
+                                "vmSize": "Standard_D4_V2"
```
Comment (logachev) [inline tools/c7n_azure/tests/templates/hdinsight.json]: Can we use different VMs? This looks like an expensive VMs and we don't run any real calculations.
Also, do we need 4 VMs?

File: tools/c7n_azure/c7n_azure/resources/hdinsight.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        policies:
+          - name: hdinsight-policy
+            resource: azure.hdinsight
+            filters:
+              - type: value
+                key: properties.computeProfile.roles[?name=='workernode'].targetInstanceCount
+                op: in
+                value: 3
+                value_type: swap
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/hdinsight.py]: Why do we need swap here?

Comment (wsf11) [reply tools/c7n_azure/c7n_azure/resources/hdinsight.py]: Used pipe to get the value

File: tools/c7n_azure/tests/templates/hdinsight.json (hunk 1 above)
Comment (wsf11) [reply tools/c7n_azure/tests/templates/hdinsight.json]: Updated to use less expensive VMs

**Pull request merged**